File: backend/test_mongo.py
import os
from pymongo import MongoClient
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

uri = os.getenv("MONGODB_URI")

print(f"Connecting to: {uri}")

try:
    client = MongoClient(uri)
    client.admin.command('ping')
    print("MongoDB connection successful!")
except Exception as e:
    print("MongoDB connection failed:", e)

File: backend/test-mongo.js
require('dotenv').config();
const mongoose = require('mongoose');

console.log('Connecting to:', process.env.MONGODB_URI); // Debug output

async function testMongoConnection() {
  try {
    await mongoose.connect(process.env.MONGODB_URI, {
      useNewUrlParser: true,
      useUnifiedTopology: true,
    });
    console.log('MongoDB connection successful!');
    await mongoose.disconnect();
    process.exit(0);
  } catch (err) {
    console.error('MongoDB connection failed:', err.message);
    process.exit(1);
  }
}

testMongoConnection();
